src/credentials/attestation_verifier/DefaultDeviceAttestationVerifier.cpp

```diff
@@ -450,15 +498,24 @@ CHIP_ERROR CsaCdKeysTrustStore::AddTrustedKey(const ByteSpan & kid, const Crypto
 
 CHIP_ERROR CsaCdKeysTrustStore::AddTrustedKey(const ByteSpan & derCertBytes)
 {
-    // TODO: Verify cert against CD root of trust (i.e. verify signatures). To do so,
-    //       we are missing primitives to validate X.509 paths of total length 2.
-
     uint8_t kidBuf[Crypto::kSubjectKeyIdentifierLength] = { 0 };
     MutableByteSpan kidSpan{ kidBuf };
     P256PublicKey pubKey;
 
     VerifyOrReturnError(CHIP_NO_ERROR == Crypto::ExtractSKIDFromX509Cert(derCertBytes, kidSpan), CHIP_ERROR_INVALID_ARGUMENT);
     VerifyOrReturnError(CHIP_NO_ERROR == Crypto::ExtractPubkeyFromX509Cert(derCertBytes, pubKey), CHIP_ERROR_INVALID_ARGUMENT);
+
+    if (!IsCdTestKey(kidSpan))
+    {
+        // Verify cert against CSA CD root of trust.
+        CertificateChainValidationResult chainValidationResult;
+        VerifyOrReturnError(CHIP_NO_ERROR ==
+                                ValidateCertificateChain(gCdRootCert, sizeof(gCdRootCert), nullptr, 0, derCertBytes.data(),
+                                                         derCertBytes.size(), chainValidationResult),
+                            CHIP_ERROR_CERT_NOT_TRUSTED);
```

Comment:
The public API states it should give CHIP_ERROR_INVALID_ARGUMENT. Either add this error to the docs in the header, or change this to CHIP_ERROR_INVALID_ARGUMENT.

Reply:
replaced with CHIP_ERROR_INVALID_ARGUMENT